plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -283,10 +295,16 @@ private static ConnectorPageSource createOrcPageSource(
                     .setBloomFiltersEnabled(options.isBloomFiltersEnabled());
             Map<IcebergColumnHandle, Domain> effectivePredicateDomains = effectivePredicate.getDomains()
                     .orElseThrow(() -> new IllegalArgumentException("Effective predicate is none"));
+
+            Optional<ReaderColumns> columnProjections = projectColumns(columns);
```

Comment:
Can we create `Optional<List<IcebergColumnHandle>>` instead of `Optional<ReaderColumns>`. It feels like only method we call on `ReaderColumns` is `get` anyway. And with that we could avoid this monster `.map(readerColumns -> readerColumns.get().stream().map(IcebergColumnHandle.class::cast).collect(toList()))` below :)

Reply:
The `ReaderProjectionsAdapter` expects `ReaderColumns` as an input, so if we're going to reuse that we need it.